src/common/stringify.ts

```diff
@@ -6,7 +6,7 @@ export function stringify(x: Node, skipTargetType = false): string {
   const result = JSON.stringify(
     x,
     (key, value) =>
-      key === "source"
+      key === "source" || key === "type"
```

Comment:
Why?

Reply:
At some point I was inserting `type: intType(low,high)` annotations from the middle of a plugin that ran before `addMutatingInfix`. But then it wasn't detecting the `i <- i+1` as a mutating infix:
https://github.com/polygolf-lang/polygolf/blob/a920b88c0403bbf4df60ed9f101856dc6599a7d2/src/plugins/ops.ts#L222-L225
Pruning out `key === "type"` is no longer needed for that use case, and I suspect you want to keep in the type annotations for elsewhere.
But if this scenario comes up, the better approach might be changing `addMutatingInfix` to not use stringify to detect an identifier with the same name --- it should just scan for `node.kind === "Identifier" && node.name === lhsName`, with a separate case for index call `a[i] <- a[i] + 1`.